cfme/tests/services/test_rest_services.py

```diff
@@ -425,6 +425,49 @@ def _finished(service):
                 delay=5
             )
 
+    @pytest.mark.parametrize(
+        "from_detail", [True, False], ids=["from_detail", "from_collection"]
+    )
+    @test_requirements.rest
+    def test_service_retirement_methods(self, request, appliance, services, from_detail):
+        """Test retiring a service with old method `retire` and new method `request_retire`.
+        Old method is no longer supported and it puts the service into `intializing` state.
+
+        Metadata:
+            test_flag: rest
+
+        Polarion:
+            assignee: pvala
+            casecomponent: Services
+            initialEstimate: 1/4h
+
+        Bugzilla:
+            1698480
+            1713477
+        """
+        service = services[0]
```

Comment:
Q: are you sure it will not raise `KeyError`?

Reply:
I am pretty sure it will not raise `KeyError`. `services` is a fixture that will create 3 services, even if there is an error, it will be caught before the test begins.